var/spack/repos/builtin/packages/rivet/package.py

```diff
@@ -72,8 +75,15 @@ class Rivet(AutotoolsPackage):
 
     depends_on("hepmc", when="hepmc=2")
     depends_on("hepmc3", when="hepmc=3")
+    # The fix for working around patch-level zero issues have landed in 4.0.1
+    # and have also been back-ported to 3.1.11. Hence we need to exclude 4.0.0
+    # explicitly and catch the rest with versions up to 3.1.10
+    # See: https://gitlab.com/hepcedar/rivet/-/merge_requests/904
+    # and: https://gitlab.com/hepcedar/rivet/-/merge_requests/912
     conflicts(
-        "hepmc@3.3.0", when="@:4.0.0 hepmc=3", msg="patch-level zero requires at least 4.0.1"
+        "hepmc@3.3.0",
```

Comment:
I think this needs a caret
```suggestion
        "^hepmc@3.3.0",
```

Reply:
Curiously both seem to work, with and without the caret, I get:
```console
$ spack solve -Il rivet@3.1.10 hepmc=3 ^hepmc@3.3.0
==> Error: failed to concretize `rivet@3.1.10 hepmc=3 ^hepmc@3.3.0` for the following reasons:
     1. Cannot satisfy 'hepmc@3.3.0'
     2. rivet: patch-level zero requires at least 3.1.11 or 4.0.1
     3. Cannot satisfy 'hepmc@3.3.0'
        required because rivet@3.1.10 hepmc=3 ^hepmc@3.3.0 requested explicitly 
     4. rivet: patch-level zero requires at least 3.1.11 or 4.0.1
        required because conflict constraint @:3.1.10,4.0.0 hepmc=3 
          required because rivet@3.1.10 hepmc=3 ^hepmc@3.3.0 requested explicitly 
        required because conflict is triggered when hepmc@3.3.0 
          required because rivet@3.1.10 hepmc=3 ^hepmc@3.3.0 requested explicitly 
```

Changed it to the caret version because that more clearly denotes a conflict with a dependency, IMO.

Reply:
Yeah, I also noticed this is still working without the caret, but I agree on the clarity argument, so thanks for changing it.